router/batchrouter/asyncdestinationmanager/yandexmetrica/yandexmetrica.go

```diff
@@ -0,0 +1,377 @@
+	ClientID any     `json:"ClientId"`
+	YclID    any     `json:"Yclid"`
+	UserID   any     `json:"UserId"`
+	Target   string  `json:"Target"`
+	DateTime string  `json:"DateTime"`
+	Price    float64 `json:"Price"`
+	Currency string  `json:"Currency"`
+}
+
+type yandexMetricaMessage struct {
+	Message yandexMetricaMessageBody `json:"message"`
+}
+
+type idStruct struct {
+	id         string
+	clientType string
+	headerName string
+}
+
+func (ym yandexMetricaMessageBody) ID() (idStruct, error) {
+	switch {
+	case ym.ClientID != nil:
+		_, ok := ym.ClientID.(string)
+		if !ok {
+			return idStruct{}, fmt.Errorf("non-string data for ClientID is not supported")
+		}
+		return idStruct{id: ym.ClientID.(string), clientType: idClientMap["ClientId"], headerName: "ClientId"}, nil
+	case ym.YclID != nil:
+		_, ok := ym.YclID.(string)
+		if !ok {
+			return idStruct{}, fmt.Errorf("non-string data for Yclid is not supported")
+		}
+		return idStruct{id: ym.YclID.(string), clientType: idClientMap["Yclid"], headerName: "Yclid"}, nil
+	case ym.UserID != nil:
+		_, ok := ym.UserID.(string)
+		if !ok {
+			return idStruct{}, fmt.Errorf("non-string data for UserId is not supported")
+		}
+		return idStruct{id: ym.UserID.(string), clientType: idClientMap["UserId"], headerName: "UserId"}, nil
+	default:
+		return idStruct{}, fmt.Errorf("no valid id found in message object")
+	}
+}
+
+type YandexMetricaBulkUploader struct {
+	logger          logger.Logger
+	Client          *http.Client
+	destinationInfo *oauthv2.DestinationInfo
+}
+
+func NewManager(destination *backendconfig.DestinationT, backendConfig backendconfig.BackendConfig) (*YandexMetricaBulkUploader, error) {
+	destinationInfo := &oauthv2.DestinationInfo{
+		Config:           destination.Config,
+		DefinitionConfig: destination.DestinationDefinition.Config,
+		WorkspaceID:      destination.WorkspaceID,
+		DefinitionName:   destination.DestinationDefinition.Name,
+		ID:               destination.ID,
+	}
+	yandexUploadManager := &YandexMetricaBulkUploader{
+		destinationInfo: destinationInfo,
+		logger:          logger.NewLogger().Child("batchRouter").Child("AsyncDestinationManager").Child("YandexMetrica").Child("YandexMetricaBulkUploader"),
+	}
+	cache := oauthv2.NewCache()
+	optionalArgs := &oauthv2httpclient.HttpClientOptionalArgs{
+		Logger:    yandexUploadManager.logger,
+		Augmenter: augmenter.YandexReqAugmenter,
+	}
+	originalHttpClient := &http.Client{Transport: &http.Transport{}}
+	// This client is used for uploading data to yandex metrica
+	yandexUploadManager.Client = oauthv2httpclient.NewOAuthHttpClient(
+		originalHttpClient,
+		oauthv2common.RudderFlowDelivery,
+		&cache,
+		backendConfig,
+		augmenter.GetAuthErrorCategoryForYandex,
+		optionalArgs,
+	)
+	return yandexUploadManager, nil
+}
+
+// return a success response for the poll request every time by default
+func (ym *YandexMetricaBulkUploader) Poll(pollInput common.AsyncPoll) common.PollStatusResponse {
```

Comment:
I have updated the async destination interfaces: https://github.com/rudderlabs/rudder-server/pull/4601/files#diff-51debce70a25da5581f2747f02baf721816436b15c2f3512c96aca5cc51c3f16

Reply:
As discussed, going ahead with the current changes for now